src/target/imxrt.c

```diff
@@ -227,6 +278,70 @@ bool imxrt_probe(target_s *const target)
 	return true;
 }
 
+static void imxrt_id_chip(target_s *target, imxrt_priv_s *priv) {
+	// The iMX RT series doesn't have a device id register. Instead, the NXP universal flash loader
+	// uses known ROM values at a particular address to differentiate the devices. That code uses
+	// three locations but only one location is actually needed.
+	// https://github.com/nxp-mcuxpresso/i.mxrt-ufl/blob/main/src/ufl_find_target.c
+	uint32_t rom_location = 0;
+	uint32_t scb_cpuid = target_mem_read32(target, SCB_CPUID);
+	uint32_t cpu_partno = (scb_cpuid & SCB_CPUID_PARTNO_Msk) >> SCB_CPUID_PARTNO_Pos;
+	if (cpu_partno == CORE_ID_CM33) {
+		rom_location = IMXRT_X00_ROM_FINGERPRINT_ADDR;
+	} else if (cpu_partno == CORE_ID_CM7) {
+		if (target->part_id == 0x88c6U) {
+			rom_location = IMXRT_11XX_ROM_FINGERPRINT_ADDR;
+		} else {
+			rom_location = IMXRT_10XX_ROM_FINGERPRINT_ADDR;
+		}
+	} else {
+		DEBUG_ERROR("Unknown core\n");
+	}
```

Comment:
These are excess braces, please remove them on all bar the else-if for the Cortex-M7 ID block